### PR TITLE
[MIG] sale_order_carrier_auto_assign: Migration to 16.0

### PR DESCRIPTION
**Pull request by matiasperalta1**

### CODE REVIEW

**Review by florenciafrigieri2 — APPROVED** [review]
Functional review ok

**Review by mmequignon — APPROVED** [review]

File: sale_order_carrier_auto_assign/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from odoo import models
+from odoo.tools import config
+
+
+class SaleOrder(models.Model):
+
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        if config["test_enable"] and not self.env.context.get(
```
Comment (rousseldenis) [inline sale_order_carrier_auto_assign/models/sale_order.py:12]: @matiasperalta1 @grindtildeath I don't like such code. Better to add a setting to enable the behaviour that breaks standard one. But not blocking

Comment (rousseldenis) [reply sale_order_carrier_auto_assign/models/sale_order.py:12]: @matiasperalta1 Could you put at least in ROADMAP we need to add a configuration setting ?

Comment (santostelmo) [reply sale_order_carrier_auto_assign/models/sale_order.py:12]: @rousseldenis I mplemented that in another PR:
https://github.com/OCA/sale-workflow/pull/2837

Comment (StephaneMangin) [reply sale_order_carrier_auto_assign/models/sale_order.py:12]: Why adding this in an other PR ? Would you not prefer to include it here ?

Comment (santostelmo) [reply sale_order_carrier_auto_assign/models/sale_order.py:12]: I would prefer. But probably @matiasperalta1 did not had opportunity to implement that.

**Pull request merged**